src/v/utils/string_switch.h

```diff
@@ -182,7 +182,10 @@ class string_switch {
 
     constexpr operator R() { // NOLINT
         if (!result) {
-            throw std::runtime_error("Fell off the end of a string-switch");
+            throw std::runtime_error(
+              std::string(
+                "Fell off the end of a string-switch while matching: ")
+              + std::string(view));
```

Comment:
nit: Maybe use `fmt::format`?

Reply:
@BenPope - it was my first instinct too, but this header didn't already include it, it includes only stdlib stuff, so I was reluctant to pull in this beast for this single string concatenation. WDYT?

Reply:
It just looks so allocation heavy these days! I don't have a strong opinion either way.

Reply:
It's in error handling, usually it's the last thing RP does before dying actually so it's probably fine :).